include/Zycore/Types.h

```diff
@@ -333,6 +333,84 @@ typedef char* ZyanCharPointer;
  */
 typedef const char* ZyanConstCharPointer;
 
+/* ============================================================================================== */
+/* Type conversion utils                                                                          */
+/* ============================================================================================== */
+
+/**
+ * Collection of functions for swapping byte order in 16/32/64-bit values respectively.
+ *
+ * @param   x       16/32/64-bit value.
+ *
+ * @return  16/32/64-bit value with byte order swapped.
+ */
+#if defined(ZYAN_MSVC)
+#   if defined(ZYAN_NO_LIBC)
+#ifdef __cplusplus
+extern "C" {
+#endif
```

Comment:
Nit: let's consistently also indent this and use `#if defined(..)`. Same for the `}` a few lines below.

Reply:
I always struggle to decide how to mix preprocessor and code indentation levels. In fixed version I went with "code starts at last preprocessor level".

Reply:
Haha, I'm also not sure if we have a consistent convention for that across the codebase.